sway-lib-std/src/storage.sw

```diff
@@ -106,3 +109,189 @@ impl<K, V> StorageMap<K, V> {
         get::<V>(key)
     }
 }
+
+/// A persistant vector struct
+pub struct StorageVec<V> {}
+
+impl<V> StorageVec<V> {
+    /// Appends the value to the end of the vector
+    /// 
+    /// # Arguments
+    /// 
+    /// * `value` - The item being added to the end of the vector
+    #[storage(read, write)]
+    pub fn push(self, value: V) {
+        // The length of the vec is stored in the __get_storage_key() slot
+        let len = get::<u64>(__get_storage_key());
+        
+        // Storing the value at the current length index (if this is the first item, starts off at 0)
+        let key = sha256((len, __get_storage_key()));
```

Comment:
Can you abstract out the getting/setting of the length into a function? As-is, the logic is repeated a few times.

Reply:
There is already a len function, but as of right now you cant call self.function()